ptp/ptp4u/server/server.go

```diff
@@ -106,6 +108,31 @@ func (s *Server) Start() error {
 		}
 	}()
 
+	// Drain check
+	go func() {
+		defer wg.Done()
+		for {
+			<-time.After(s.Config.DrainInterval)
+
+			var drain bool
+			for _, check := range s.Checks {
+				if check.Check() {
+					drain = true
+					break
```

Comment:
I would print here "Check X failed" and don't `break` but continue checks anyway.

Reply:
I personally love it see "is it one only failing or all of them". But no strong feelings on this part.